src/nom.rs

```diff
@@ -0,0 +1,201 @@
+            }
+            Lambda | Lambda2 | Lambda3 => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for MozjsCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Mozjs::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for JavascriptCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Javascript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TypescriptCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Typescript::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for TsxCode {
+    fn compute(node: &Node, _code: &[u8], stats: &mut Stats) {
+        use Tsx::*;
+
+        match node.kind_id().into() {
+            Function | FunctionDeclaration | MethodDefinition => {
+                stats.functions.insert(std::string::String::new());
+            }
+            GeneratorFunction | GeneratorFunctionDeclaration | ArrowFunction => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for RustCode {
+    fn compute(node: &Node, code: &[u8], stats: &mut Stats) {
+        use Rust::*;
+
+        match node.kind_id().into() {
+            FunctionSignatureItem | FunctionItem => {
+                if let Some(id) = node.child(1) {
+                    let code = &code[id.start_byte()..id.end_byte()];
+                    let string_code = String::from_utf8(code.to_vec()).unwrap();
+                    stats.functions.insert(string_code);
+                }
+            }
+            ClosureExpression => {
+                stats.closures += 1;
+            }
+            _ => {}
+        }
+    }
+}
+
+impl Nom for CppCode {
+    fn compute(node: &Node, code: &[u8], stats: &mut Stats) {
+        use Cpp::*;
+
+        match node.kind_id().into() {
+            FunctionDeclarator | FunctionDeclarator2 | FunctionDeclarator3 => {
```

Comment:
Because of polymorphism it's probably wrong here:
```cpp
struct A {
  void foo(int) {}
  void foo(double) {}
};
```
The 2 functions have the same name but are different